### PR TITLE
fix: Align code with flash_api.cpp when num_splits > 1

### PR DESCRIPTION
**Pull request by Seventeen17**

### CODE REVIEW

File: torch_xla/csrc/ops/flash_attention_forward.cpp [hunk 1]
```diff
@@ -235,13 +235,25 @@ void custom_call_flash_attention_forward(cudaStream_t stream, void** buffers,
...
       launch_params.num_splits =
           num_splits_heuristic(params.b * params.h * num_m_blocks,
                                dprops->multiProcessorCount, num_n_blocks, 128);
     }
+    if (launch_params.num_splits > 1) {
+      at::Tensor softmax_lse_accum =
+          torch::empty({launch_params.num_splits, params.b, params.h,
+                        launch_params.seqlen_q},
+                       opts.dtype(at::kFloat));
+      at::Tensor out_accum =
```
Comment (yitongh) [inline torch_xla/csrc/ops/flash_attention_forward.cpp]: Will these two tensors be destructed once they leave the scope of the if statement?

File: torch_xla/csrc/ops/flash_attention_forward.cpp [hunk 2]
```diff
@@ -235,13 +235,25 @@ void custom_call_flash_attention_forward(cudaStream_t stream, void** buffers,
   // inference.
   const int num_m_blocks = (params.seqlen_q + 64 - 1) / 64;
   launch_params.num_splits = 0;
-  if (1 - params.p_dropout == 0.0f) {  // SplitKV is not implemented for dropout
+  if (params.p_dropout == 1.0f) {  // SplitKV is not implemented for dropout
```
Comment (anw90) [inline torch_xla/csrc/ops/flash_attention_forward.cpp:238]: Is that dangerous to use `==` to compare a float value. Maybe we could use `fabs(1-params.p_dropout) < 1e-6` to check the dropout value.

File: torch_xla/csrc/ops/flash_attention_forward.cpp [hunk 3]
```diff
@@ -235,13 +235,25 @@ void custom_call_flash_attention_forward(cudaStream_t stream, void** buffers,
...
     if (launch_params.num_splits < 1) {
```
Comment (yitongh) [inline torch_xla/csrc/ops/flash_attention_forward.cpp:239]: Should be params.num_splits here? `launch_params.num_splits = 0` in L237.

File: torch_xla/csrc/ops/flash_attention_forward.cpp (hunk 1 above)
Comment (Seventeen17) [reply torch_xla/csrc/ops/flash_attention_forward.cpp]: You're right, I've modified it.

File: torch_xla/csrc/ops/flash_attention_forward.cpp (hunk 3 above)
Comment (Seventeen17) [reply torch_xla/csrc/ops/flash_attention_forward.cpp:239]: Done

File: torch_xla/csrc/ops/flash_attention_forward.cpp (hunk 2 above)
Comment (Seventeen17) [reply torch_xla/csrc/ops/flash_attention_forward.cpp:238]: Done. Although the original impl in FA use `==` but yes, you are right.

**Review by anw90 — APPROVED** [review]

**Review by yitongh — APPROVED** [review]

**Pull request merged**